### ISSUE TITLE
Wkhtmltopdf SVG rendering issues

### ISSUE THREAD

**Issue by maddogcz** [wkhtmltopdf/wkhtmltopdf#4611]
Hi,

I'm trying to use wkhtmltopdf for creating reports in PDF format by converting very simple HTML with just <img> tags pointing to static svg files (no JS at all).

When I use a version from my desktop Ubuntu 18.10 repo (0.12.4 WITHOUT patched QT) it works fine except it cannot handle more than 1 file (which is a limitation I can overcome).
When I use any (.deb) version from GIT witch patched QT, I run into great number of problems such as:

- not rendering some paths witch clip-path set
- ignoring alpha property (whilst specifying fill color with alpha works fine like fill="#FF00AA30")

Problem is, I need to use in on Centos 7 server, where repo version works only with virtual X server (XVFB)  which rasterizes all <img>s, so It's pretty useless. Is there some way I can get the unpatched QT version for centos?

**This is rendered with unpatched 0.12.4 from repo**
![wkhtml-0 12 4](https://user-images.githubusercontent.com/7744283/74166536-020c0880-4c27-11ea-8714-6c45b01933dc.png)

**This is 0.12.5 from .deb package**
![wkhtml-0 12 5_patched_QT](https://user-images.githubusercontent.com/7744283/74166534-01737200-4c27-11ea-8a0b-306190e929b3.png)

**Comment by PhilterPaper** [comment]
I think wkHTMLtoPDF will successfully convert (most) SVG inline with the HTML (`<svg>` tag), but I don't think I've ever seen anyone use `<img>` tags pointing to SVG files. Do you have any control over the HTML files you're using, and could try direct SVG inlining? Perhaps current HTML browsers are able to use SVG files in `<img>` tags, but WebKit is fairly old and might not be able to do that.

**Comment by maddogcz** [comment]
Thanks, I tried to make the SVG inline, but with no effect. 

I've attached some minimal example (matplotlib generated SVG plot) that causes trouble, so if you or someone please look into this. Thank you.

[svg.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/4184463/svg.zip)

**Comment by PhilterPaper** [comment]
It looks like this can be fixed with two changes to the SVG code (embedded in HTML... I have no idea if you can use `<img>` src'ing an .svg file). These may be considered just workarounds for a couple of glitches in WebKit, since the original works OK with a browser:
1. In the code to draw the colored (very light red) band, I don't think the opacity is being honored. Try changing `" style="fill:#ff0000;opacity:0.1;stroke:#ff0000;stroke-linejoin:miter;"/>` to `" style="fill:#e0ffff;stroke:#e0fff;stroke-linejoin:miter;"/>`.
2. In the code to draw the dashed line through the red band, something is going wrong with the clip-path. You should be able to remove it: for `id="line2d_11"`, change `<path clip-path="url(#p0abd9c53e4)" d="M 57.6 86.066919` to `<path d="M 57.6 86.066919`. The use of this clip-path elsewhere doesn't seem to be causing problems.

Are you aware that your SVG program is individually drawing each character with lines and splines, rather than using SVG's `<text>` tag? Most of your file is taken up with this.

**Comment by maddogcz** [comment]
Yes, for these two particular issues I am able to use workaround, but sometimes the clip-path is necessary (i.e. when error bars get outside the plot)

Text rendering may be changed (or not), as I said, the SVG is generated by matplotlib.

Nevertheless the QT-unpatched version of wkhtml have no such issues, the SVG rendering is perfect...

**Comment by PhilterPaper** [comment]
As I mentioned, the clip path didn't seem to be causing problems elsewhere. I did notice one of the error bars slightly peeping above the top axis, so maybe the clip path is off a few pixels (removing the dashed line in the process). It might be worth playing with the coordinates of the clip path. The opacity is the other thing -- does that work in the unpatched QT?

**Comment by maddogcz** [comment]
Clip-path is disabled for the error bars, otherwise, they wouldn't be rendered at all.
I prepared some better examples

[wktest.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/4191396/wktest.zip)

**Comment by PhilterPaper** [comment]
wkHTMLtoPDF 0.12.5 with patched QT, again I have to remove opacity (and change the color) on the filled bar, and have to remove the clip-path for the straight dashed line. However, this time the error bars have disappeared. Removing all the clip-path attributes brought them back -- maybe your library is just overdoing it with clipping? Maybe don't do clipping unless you see that content is outside of reasonable bounds, an then add some manual clipping as needed? Or, if using the unpatched wkHTMLtoPDF does the job for you, that's fine, too.

**Comment by maddogcz** [comment]
Well the best solution for now would be to get my hands on unpatched wkHTMLtoPDF in form of .rpm package since the repo version doesn't work well on server for me...

**Comment by PhilterPaper** [comment]
See also #4232
